src/main/scala/viper/silver/plugin/standard/reasoning/analysis/VarAnalysisGraphMap.scala

```diff
@@ -0,0 +1,390 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2024 ETH Zurich.
+
+package viper.silver.plugin.standard.reasoning.analysis
+
+import viper.silver.ast.utility.Expressions
+import viper.silver.ast.{AccessPredicate, Apply, Assert, Assume, BinExp, CurrentPerm, Declaration, DomainFuncApp, Exhale, Exp, FieldAccess, FieldAssign, Fold, ForPerm, FuncApp, Goto, If, Inhale, Label, LocalVar, LocalVarAssign, LocalVarDecl, Method, MethodCall, Package, Position, Program, Quasihavoc, Quasihavocall, Ref, Scope, Seqn, Stmt, UnExp, Unfold, While}
+import viper.silver.plugin.standard.reasoning.analysis.AnalysisUtils.{AssumeNode, getLocalVarDeclFromLocalVar, heapVertex}
+import viper.silver.plugin.standard.reasoning.{Assumes, ExistentialElim, FlowAnnotation, FlowVar, Heap, NoAssumeAnnotation, OldCall, UniversalIntro, Var}
+import viper.silver.plugin.standard.termination.{DecreasesSpecification, DecreasesStar, DecreasesTuple, DecreasesWildcard}
+import viper.silver.verifier.{AbstractError, ConsistencyError}
+
+import scala.collection.mutable
+
+object AnalysisUtils {
+
+  // TODO Add NoAssume Specification (default for all cases except for non-terminating abstract methods (e.g AssumeFree).
+  //      Also check method Body for assumes (or inhales in case this specification is present
+  case class AssumeNode(pos: Position) extends Declaration {
+    override def name: String = ".assume"
+  }
+
+  val heapVertex: LocalVarDecl = LocalVarDecl(".heap", Ref)()
+
+  def getDeclarationFromFlowVar(f: FlowVar, m: Method): Declaration = {
+    f match {
+      case v: Var => LocalVarDecl(v.decl.name, v.decl.typ)(v.decl.pos)
+      case _: Heap => heapVertex
+      case _: Assumes => AssumeNode(m.pos)
+    }
+  }
+
+  def getLocalVarDeclFromFlowVar(f: FlowVar): LocalVarDecl = {
+    f match {
+      case v: Var => LocalVarDecl(v.decl.name, v.decl.typ)(v.decl.pos)
+      case _: Heap => heapVertex
+    }
+  }
+
+  def getLocalVarDeclFromLocalVar(l: LocalVar): LocalVarDecl = {
+    LocalVarDecl(l.name, l.typ)()
+  }
+
+  def containsDecreasesAnnotations(m: Method): (Boolean, Boolean) = {
+    val presContain = m.pres.collect({ case DecreasesTuple(_, _) | DecreasesWildcard(_) | DecreasesStar() => true }).nonEmpty
+    val postsContain = m.posts.collect({ case DecreasesTuple(_, _) | DecreasesWildcard(_) | DecreasesStar() => true }).nonEmpty
+
+    val presContainStar = m.pres.collect({ case DecreasesStar() => true }).nonEmpty
+    val postsContainStar = m.posts.collect({ case DecreasesStar() => true }).nonEmpty
+
+
+    /** check info for decreases specification */
+    val infoContains = m.meta._2 match {
+      case _: DecreasesSpecification => true
+      case _ => false
+    }
+
+    val infoContainsStar = m.meta._2 match {
+      case spec: DecreasesSpecification => spec.star.isDefined
+      case _ => false
+    }
+
+    val containsTerminationMeasure = presContain | postsContain | presContainStar | postsContainStar | infoContains
```

Comment:
Isn't `infoContainsStar` missing?
```suggestion
    val containsTerminationMeasure = presContain | postsContain | presContainStar | postsContainStar | infoContains | infoContainsStar
```